### PR TITLE
NO-JIRA Update azdo task versions

### PR DESCRIPTION
**Pull request by lucas-paulger-sonarsource**
Please review our [contribution guidelines](https://github.com/SonarSource/sonar-scanner-azdo/blob/master/docs/contributing.md).

And please ensure your pull request adheres to the following guidelines: 

- [ ] Please explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [ ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [ ] Provide a unit test for any code you changed
- [ ] If there is a [JIRA](https://jira.sonarsource.com/browse/SONARAZDO) ticket available, please make your commits and pull request start with the ticket ID (SONARAZDO-XXXX)

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -151,24 +145,6 @@ stages:
...
-                extension_installed=""
-                while [ -z "$extension_installed" ]; do
-                  echo "Waiting for SonarCloudPublishTest task to be registered..."
-                  tasks=$(tfx build tasks list --authType pat --service-url "$SERVICE_URL" -t "$AZURE_TOKEN" --no-color --json)
-                  extension_installed=$(echo $tasks | jq -r ".[] | select(.name == \"SonarCloudPublishTest\") | .id")
-                  if [ -z "$extension_installed" ]; then
-                    sleep 5
-                  fi
-                done
-                echo "SonarCloudPublishTest task is registered with ID: $extension_installed"
```
Comment (7PH) [inline azure-pipelines.yml:171]: I reckon that now that we're not uninstalling the extension anymore, that's not needed 🤔

File: azure-pipelines.yml [hunk 2]
```diff
@@ -106,28 +106,22 @@ stages:
...
             inputs:
               version: "v0.10.x"
 
           # Publish Extension
-          - task: TfxInstaller@3
-            displayName: "Install TFX CLI"
-            env:
-              NPM_CONFIG_//repox.jfrog.io/artifactory/api/npm/:_authToken: $(ARTIFACTORY_PRIVATE_READER_ACCESS_TOKEN)
-            inputs:
-              version: "v0.10.x"
```
Comment (7PH) [inline azure-pipelines.yml:122]: 👍

**Review by 7PH — APPROVED** [review]
LGTM

File: azure-pipelines.yml (hunk 1 above)
Comment (lucas-paulger-sonarsource) [reply azure-pipelines.yml:171]: exactly, because the tasks are always there; idk if there's a way to wait for the new version of a task to be extra sure

**Comment by 7PH** [comment]
I've just made an unrelated test and noticed that just after installing the extension, without dropping/re-installing it, a mix of multiple versions were used in my pipeline:

```
Task         : Prepare Analysis Configuration
Description  : Prepare SonarQube Cloud analysis configuration
Version      : 3.0.803

Task         : Run Code Analysis
Description  : Run scanner and upload the results to the SonarCloud server.
Version      : 3.0.802
```

So this may be something we want to ensure in our pipeline? At least I'll manually check that it doesn't happen for the next 2-3 builds

**Pull request merged**